doc/iterations/2023.10/shapes/contribute_custom_frontend_tool.adoc

```diff
@@ -0,0 +1,65 @@
+= (S) Contribute custom frontend tool
+
+== Problem
+
+Today there is no way to add custom complex tool to a diagram palette.
+It's only possible to add semantic actions on a clickHandler, but not to add more complex scenarios like open a modal to select data.
+
+== Key Result
+
+Add the possibility to programmatically contribute custom React Components tools to a palette.
+
+== Solution
+
+Add a new React component `DiagramPaletteToolContribution` with two props:
+
+* `canHandle(diagramId: string, diagramElementData: DiagramElementData)`->defined if the custom tool must be added to the palette.
+`DiagramElementData` is a union type, it'll depend on if it's a diagram palette or a node palette.
+** For a diagram palette, the programmer will have access to the diagram metadata:
+
+[source]
+----
+  id: string;
+  kind: string;
```

Comment:
Won't this always be `siriusComponents://representation?type=Diagram`, and so not really useful?

Reply:
It's part of the object `DiagramMetadata` that I reused for the props.